src/main/java/com/faforever/client/FafClientApplication.java

```diff
@@ -87,7 +88,7 @@ public void init() {
     Font.loadFont(FafClientApplication.class.getResourceAsStream("/font/dfc-icons.ttf"), 10);
     JavaFxUtil.fixTooltipDuration();
 
-    applicationContext = new SpringApplicationBuilder(FafClientApplication.class)
+    applicationContext = new SpringApplicationBuilder(FafClientApplication.class, ForgedAlliancePrefs.class)
```

Comment:
Why is thatdone?

Reply:
Without that there is no Bean found for ForgedAlliancePrefs, which is needed in the MapService. 

Possible alternative: undo that change, make field 'forgedAlliancePreferences' in MapService non-final and set it inside afterPropertiesSet() (--> forgedAlliancePreferences = preferencesService.getPreferences().getForgedAlliance();). (Not possible to do that on field declaration as the injected preferencesService is needed here).

I changed it to the latter, please have a look at my new commit and see if it is the better approach.

Reply:
ForgedAlliancePrefs should ot be registered with Spring it is retrived via the preferenceService

Reply:
There is a huge difference between the two ways :D The one (preferenceService) loades data from a json for settings and the other just creates a new instance(Spring)

Reply:
Also the way to add Beans to Spring is to add @Component to the class and not this